chanzuckerberg/edu-design-system
# feat(typography): split into Heading and Text components

### Summary:
[[ch119356]](https://app.clubhouse.io/czi-edu/story/119356/ds-split-typography-into-heading-and-text-components)

From [discussion](https://docs.google.com/document/d/1qlnMcX03slmFYsCspAOxo6lx2posZvhemJdkAzLzdlk/edit?ts=601b1f47#), there are a few benefits to this:
- heading and non-heading text are different semantic use cases
- leaves the door open for something like a HeadingContext in the future
- traject is already used to a Heading component
- it's a pretty common DS pattern

I also removed Typography from the playground button and capitalized the file names

### Test Plan:
- Confirm snapshots look the same as before (just split across 2 files)
- Confirm no button changes in Percy (only visual changes are from new Heading/Text file names)
- Confirm docs look good
